peers/app_request_network.go

```diff
@@ -94,38 +94,47 @@ func NewNetwork(
 		return nil, err
 	}
 
-	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.PChainAPI)
+	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.GetPChainAPI())
 
 	arNetwork := &AppRequestNetwork{
-		Network:         testNetwork,
-		Handler:         handler,
+		network:         testNetwork,
+		handler:         handler,
 		infoAPI:         infoAPI,
 		logger:          logger,
 		lock:            new(sync.Mutex),
 		validatorClient: validatorClient,
 	}
+	go logger.RecoverAndPanic(func() {
+		testNetwork.Dispatch()
+	})
+
+	return arNetwork, nil
+}
 
+// TODO: remove dependence on Relayer specific config since this is meant to be a generic AppRequestNetwork file
+func (n *AppRequestNetwork) InitializeConnectionsAndCheckStake(cfg *config.Config) error {
 	// Manually connect to the validators of each of the source subnets.
 	// We return an error if we are unable to connect to sufficient stake on any of the subnets.
 	// Sufficient stake is determined by the Warp quora of the configured supported destinations,
 	// or if the subnet supports all destinations, by the quora of all configured destinations.
 	for _, sourceBlockchain := range cfg.SourceBlockchains {
 		if sourceBlockchain.GetSubnetID() == constants.PrimaryNetworkID {
-			if err := arNetwork.connectToPrimaryNetworkPeers(cfg, sourceBlockchain); err != nil {
-				return nil, err
+			if err := n.connectToPrimaryNetworkPeers(cfg, sourceBlockchain); err != nil {
+				return fmt.Errorf(
+					"Failed to connect to primary network peers: %w",
```

Comment:
when returning error with `fmt.Errorf` we use lower case error mesasge, when printing/logging a new error we capitalize